azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -160,20 +176,20 @@ def update_host_plugin_headers(wire_client):
         # Fetching the goal state updates the HostGAPlugin so simply trigger the request
         GoalState._fetch_goal_state(wire_client)
 
-    def update(self, silent=False):
+    def update(self, goalstate_properties=GoalStateProperties.default_properties(), silent=False):
```

Comment:
The update method should probably not take the properties as parameter, since it opens the possibility of passing a value different to the one passed to init() and that would lead to an inconsistent goal state.

Reply:
This has been removed